### PR TITLE
add kornia data augmentation tutorial

### PR DESCRIPTION
**Pull request by edgarriba**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown] id="x83-rnVKT8Wo"
+# # GPU and batched data augmentation with Kornia and PyTorch-Lightning
+#
+# In this tutorial we will show how to combine both [Kornia.org](https://www.kornia.org) and PyTorch Lightning to perform efficient data augmentation to train a simpple model using the GPU in batch mode without additional effort.
```
Comment (Borda) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:20]: this goes to meta

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
..."599c941c2a5748a1bd23c01997f89deb", "1159393a62f8445b9987cc5c6a0345b9", "0aa995b993044fc2859a0bd78bd0f574", "3435453a01004b8ab13aba41380669c0", "f1f82dafccc94d36b94235edc5e8e9a2", "e4ada31d4fa247d09f4902d24c0de8fc", "eba24180d9504770a215579e231f77c5", "fe78a1c28f4e48e580e90bb86c2cd974", "de0a418bdb7b4645a720e42c4360bc56", "5dc59cbf46d34facbf6bda28d901f23d", "77f480b972fa4a429985b665a703bca5", "1ad61d6d3ba742058f33ba42d0f7aee5", "40ac1628402c404397f5f7fe4a392ec4", "58907965ba6548698fd4ee09d8cc537f", "c6173d0bae434ad894bbcdb950978597", "5a19c9bd4631496c83c086c9a0790eb7", "36bc9e0864c041a3a9ba99138526dd0a", "9e5951cd14134a01a0989847610e5158", "b2ab723a9bd442caa234c31c3a0b37f2", "ba95f0a20a644817838d65454c68e62a", "6d1131a36b0643d0bbe361b1dce2530a", "2aad354dd58a4370ab02e9717b222527", "307eda2be3d44b6586e10cde67457eb1", "a86b460ff8b64039b3c1ad9c20fc509b", "dd8c0b79bc1e4543827f4c945e612afe", "98b00e759bd1480d8642e74dd513bb81", "8a1bc0325fa340c09908f90fdc5c3d3b", "0db90d9f3f8f4b369c902c4e1fb7d872", "6ec62f397aa24e08ae2d3333333e67e2", "6ad4eb9da10b4257aaae2edc87ed75bd", "013b59fa0c4d487eb71f900a4703c7bc", "aa6bdb26784c46e98dadcd8ccef5af45"]} outputId="12d1be70-3b5c-41e1-95bd-1f9d91bd07ae"
```
Comment (Borda) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:196]: what is this?

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+# Initialize a trainer
+trainer = Trainer(
+    progress_bar_refresh_rate=20,
+    gpus=min(1, torch.cuda.device_count()),
```
Comment (Borda) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:200]: lest define it as come constant at the beginning

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py (hunk 2 above)
Comment (edgarriba) [reply starters/augmentations/data_augmentation_kornia_lightning_gpu.py:196]: this was auto-generated by jupytext

**Comment by edgarriba** [comment]
@Borda can we just set one single gpu ? now the azure pipeline is failing possibly for this reason

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+import torch
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+AVAIL_GPUS: int = min(1, torch.cuda.device_count())
```
Comment (edgarriba) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:46]:
```suggestion
# AVAIL_GPUS: int = min(1, torch.cuda.device_count())
AVAIL_GPUS: int = 1
```

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+# AVAIL_GPUS: int = min(1, torch.cuda.device_count())
+AVAIL_GPUS: int = 1
```
Comment (edgarriba) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:47]:
```suggestion
AVAIL_GPUS = 1 if torch.cuda.is_available() else None
```

File: starters/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+import torch
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+# AVAIL_GPUS: int = min(1, torch.cuda.device_count())
```
Comment (edgarriba) [inline starters/augmentations/data_augmentation_kornia_lightning_gpu.py:46]:
```suggestion
```

**Comment by Borda** [comment]
> @Borda can we just set one single gpu ? now the azure pipeline is failing possibly for this reason

sure, I did the same in other notebooks too

**Comment by edgarriba** [comment]
@Borda green lights. Please, review again.

File: lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+# # GPU and batched data augmentation with Kornia and PyTorch-Lightning
+#
+# **NOTE:** Adaptation of the original post found in [Kornia tutorials](https://kornia-tutorials.readthedocs.io/en/latest/).
+
+# %% [markdown] colab_type="text" id="iCsre0XmawoR"
+# ## Setup
+# We first need to install Kornia and PyTorch Lightning and for convenience also torchmetrics.
+
+# %% colab={} colab_type="code" id="dEeUzX_5aLrX"
+# ! pip install kornia pytorch_lightning torchmetrics -qU
```
Comment (Borda) [inline lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py:27]:
```suggestion
```
The title is taken from metafile and requirements will be also generated #37

File: lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+import torch
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+AVAIL_GPUS = 1 if torch.cuda.is_available() else None
```
Comment (Borda) [inline lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py:46]:
```suggestion
AVAIL_GPUS = min(1, torch.cuda.device_count())
```

Comment (edgarriba) [reply lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py]: this made the cpu to hang

**Comment by edgarriba** [comment]
@Borda any idea why docs are still failing ?

File: lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py (hunk 8 above)
Comment (Borda) [reply lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py]: is this linked to PL or Kornia usage? bc it forks fine on other notebooks

**Comment by Borda** [comment]
> @Borda any idea why docs are still failing?

you need to resolve all `WARNING: Title level inconsistent` which is also what I told you here https://github.com/PyTorchLightning/lightning-tutorials/pull/28#discussion_r651286392

File: lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py (hunk 8 above)
Comment (edgarriba) [reply lightning_examples/augmentations/data_augmentation_kornia_lightning_gpu.py]: we can try again but it wasn't working yesterday

File: lightning_examples/augmentation_kornia/augmentation.py [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+import torch
+import torch.nn as nn
+import torchmetrics
+import torchvision
+from pytorch_lightning import Trainer
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+
+AVAIL_GPUS = 1 if torch.cuda.is_available() else None
```
Comment (edgarriba) [inline lightning_examples/augmentation_kornia/augmentation.py:46]:
```suggestion
AVAIL_GPUS = min(1, torch.cuda.device_count())
```

**Comment by edgarriba** [comment]
@Borda green lights again. The gpus trick worked this time - black magic :)

File: lightning_examples/augmentation_kornia/augmentation.py [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown] colab_type="text" id="iCsre0XmawoR"
+# ## Setup
+# We first need to install Kornia and PyTorch Lightning and for convenience also torchmetrics.
+
+# %% colab={} colab_type="code" id="dEeUzX_5aLrX"
+# ! pip install kornia pytorch_lightning torchmetrics -qU
```
Comment (Borda) [inline lightning_examples/augmentation_kornia/augmentation.py:22]:
```suggestion
```
this is generated

**Review by Borda — APPROVED** [review]

**Pull request merged**